gnovm/stdlibs/std/banker.go

```diff
@@ -88,10 +89,30 @@ func X_bankerTotalCoin(m *gno.Machine, bt uint8, denom string) int64 {
 
 func X_bankerIssueCoin(m *gno.Machine, bt uint8, addr string, denom string, amount int64) {
 	// gno checks for bt == RealmIssue
-	m.Context.(ExecContext).Banker.IssueCoin(crypto.Bech32Address(addr), denom, amount)
+
+	// check origin denom format
+	matched, err := regexp.MatchString("[a-z][a-z0-9]{2,15}", denom)
+	if err != nil || !matched {
+		m.Panic(typedString("invalid denom format, must be [a-z][a-z0-9]{2,15}"))
+		return
+	}
+
+	// Similar to ibc spec
+	// ibc_denom := 'ibc/' + hash('path' + 'base_denom')
+	// gno_realm_denom := '/' + 'pkg_path' + ':' + 'base_denom'
+	newDenom := "/" + m.Realm.Path + ":" + denom
+	m.Context.(ExecContext).Banker.IssueCoin(crypto.Bech32Address(addr), newDenom, amount)
```

Comment:
Just so I understand better - where do we actually limit the banker to not be able to mint GNOTs?

Reply:
> where do we actually limit the banker to not be able to mint coins under the "ugnot" denom?

We don't. I mean... realm can try to mint 'ugnot' but underhood vm won't just use raw denom input but denom with some extra string (pkg_path). So.. I'd rather say we don't limit but changes denom.

> should we generally disallow coins that have "ugnot" & "gnot" in their denom, to avoid confusion?

this IS tricky. I think it kinda relates when we support IBC. What will happen if other gno based chain's coin(ugnot) comes to gno.land over IBC? How are we going to handle its denom ??

Reply:
@moul WDYT?

Reply:
We can add a configurable blacklist system in the future, but the denom's realm path should suffice for now.